packages/ag-charts-enterprise/src/features/sync/chartSync.ts

```diff
@@ -0,0 +1,106 @@
+import { type AgChartSyncOptions, _ModuleSupport } from 'ag-charts-community';
+
+const { BOOLEAN, STRING, UNION, BaseProperties, ChartUpdateType, ObserveChanges, Validate } = _ModuleSupport;
+
+export class ChartSync extends BaseProperties implements _ModuleSupport.ModuleInstance, AgChartSyncOptions {
+    static className = 'Sync';
+
+    @Validate(BOOLEAN)
+    enabled: boolean = false;
+
+    @Validate(STRING, { optional: true })
+    @ObserveChanges<ChartSync>((target, newValue, oldValue) => {
+        const { syncManager } = target.moduleContext;
+        syncManager.unsubscribe(oldValue);
+        syncManager.subscribe(newValue);
+        target.updateSiblings(oldValue);
+        target.updateSiblings(newValue);
+    })
+    groupId?: string;
+
+    @Validate(UNION(['x', 'y', 'xy'], 'an axis'))
+    @ObserveChanges<ChartSync>((target) => {
+        const { syncManager } = target.moduleContext;
+        target.updateChart(syncManager.getChart());
+    })
+    axes: 'x' | 'y' | 'xy' = 'x';
+
+    @Validate(BOOLEAN)
+    @ObserveChanges<ChartSync>((target, newValue) => {
+        if (newValue) {
+            target.enabledZoomSync();
+        } else {
+            target.disableZoomSync?.();
+        }
+    })
+    zoom: boolean = true;
+
+    constructor(protected moduleContext: _ModuleSupport.ModuleContext) {
+        super();
+    }
+
+    private updateChart(chart: any) {
+        chart.updateService.update(ChartUpdateType.UPDATE_DATA, { skipSync: true });
+    }
+
+    private updateSiblings(groupId?: string) {
+        for (const chart of this.moduleContext.syncManager.getSiblings(groupId)) {
+            this.updateChart(chart);
+        }
+    }
+
+    private enabledZoomSync() {
+        const { syncManager, zoomManager } = this.moduleContext;
+        this.disableZoomSync = zoomManager.addListener('zoom-change', (e) => {
+            if ('stopPropagation' in e && e.stopPropagation) return;
+            for (const chart of syncManager.getSiblings(this.groupId)) {
+                chart.zoomManager.updateZoom('zoom', zoomManager.getZoom());
+                (chart.zoomManager as any).listeners.dispatch('zoom-change', { ...e, stopPropagation: true });
```

Comment:
Shouldn't `updateZoom()` percolate the change itself?

Reply:
you're right. good catch 👍